File: 3-leetcode/01-easy/js/145-binary-tree-postorder-traversal.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

const postorderTraversal = (root: TreeNode | null): number[] => {
  return traverse(root, []);
};

function traverse(node: TreeNode | null, result: number[]) {
  if (!node) {
    return result;
  }

  traverse(node.left, result);
  traverse(node.right, result);
  result.push(node.val);

  return result;
}

File: 3-leetcode/01-easy/dart/145-binary-tree-postorder-traversal.dart
class TreeNode {
  int val;
  TreeNode? left;
  TreeNode? right;
  TreeNode([this.val = 0, this.left, this.right]);
}

class Solution {
  List<int> result = [];
  List<int> postorderTraversal(TreeNode? root) {
    traverse(root);
    return result;
  }

  void traverse(TreeNode? root) {
    if (root == null) {
      return;
    }
    traverse(root.left);
    traverse(root.right);
    result.add(root.val);
  }
}
